examples/quad/main.rs

```diff
@@ -677,78 +636,39 @@ fn main() {
             // Verify that previous format still exists so we may reuse it.
             assert!(formats.iter().any(|fs| fs.contains(&format)));
 
-            let swap_config = SwapchainConfig::from_caps(&caps, format, resize_dims);
+            let swap_config = SurfaceSwapchainConfig::from_caps(&caps, format);
             println!("{:?}", swap_config);
-            let extent = swap_config.extent.to_extent();
-
-            let (new_swap_chain, new_backbuffer) =
-                unsafe { device.create_swapchain(&mut surface, swap_config, Some(swap_chain)) }
-                    .expect("Can't create swapchain");
-
             unsafe {
-                // Clean up the old framebuffers, images and swapchain
-                for framebuffer in framebuffers {
-                    device.destroy_framebuffer(framebuffer);
-                }
-                for (_, rtv) in frame_images {
-                    device.destroy_image_view(rtv);
-                }
+                surface.configure_swapchain(&device, swap_config)
+                    .expect("Can't configure swapchain");
             }
 
-            backbuffer = new_backbuffer;
-            swap_chain = new_swap_chain;
-
-            let (new_frame_images, new_framebuffers) = {
-                let pairs = backbuffer
-                    .into_iter()
-                    .map(|image| unsafe {
-                        let rtv = device
-                            .create_image_view(
-                                &image,
-                                i::ViewKind::D2,
-                                format,
-                                Swizzle::NO,
-                                COLOR_RANGE.clone(),
-                            )
-                            .unwrap();
-                        (image, rtv)
-                    })
-                    .collect::<Vec<_>>();
-                let fbos = pairs
-                    .iter()
-                    .map(|&(_, ref rtv)| unsafe {
-                        device
-                            .create_framebuffer(&render_pass, Some(rtv), extent)
-                            .unwrap()
-                    })
-                    .collect();
-                (pairs, fbos)
-            };
-
-            framebuffers = new_framebuffers;
-            frame_images = new_frame_images;
-            viewport.rect.w = extent.width as _;
-            viewport.rect.h = extent.height as _;
+            viewport.rect.w = resize_dims.width as _;
+            viewport.rect.h = resize_dims.height as _;
             recreate_swapchain = false;
+            framebuffers.clear();
```

Comment:
We ought to call `destroy_framebuffer` here on the framebuffers before dropping them.